components/calendar/calendar.js

```diff
@@ -0,0 +1,268 @@
+       * to the first position of the selectedDates array.
+       * If the selectionType is 'range', we need to verify the following requirements:
+       *
+       *   - If there's no start date selected, then the selected date becomes the start
+       * date and the minLimit becomes that same date. Prevents the range selection to the past.
+       *
+       *   - If there's a start date already selected:
+       *
+       *     - If there's no end date selected, then the selected date becomes the end date. Also
+       * if the start and end dates are the same, it will remove the minLimit as the layout renders
+       * them as a 'normal' selection.
+       *
+       *     - If there's an end date selected and the user is clicking on the start date again, it
+       * clears the selections and the limits, resetting the range.
+       */
+      if (selectionType === CALENDAR_SELECTION_TYPE_RANGE) {
+        if (selectedDates[0]) {
+          if (!selectedDates[1]) {
+            selectedDates[1] = dateSelected;
+            if (selectedDates[0].toDateString() === selectedDates[1].toDateString()) {
+              minLimit = minDate ? new Date(minDate) : null;
+            }
+          } else {
+            selectedDates = [null, null];
+            minLimit = minDate ? new Date(minDate) : null;
+          }
+        } else {
+          selectedDates[0] = dateSelected;
+          minLimit = dateSelected;
+          selectedDates[1] = null;
+        }
+      } else {
+        selectedDates[0] = dateSelected;
+      }
+
+      /**
+       * If the user selects a day of the previous or next month, the rendered month switches to
+       * the one of the selected date.
+       */
+      if (dateSelected.getMonth() !== renderDate.getMonth()) {
+        renderDate = new Date(dateSelected.toDateString());
+      }
+
+      return {
+        minLimit,
+        renderDate,
+        selectedDates,
+      };
+    }, () => {
+      if (onSelectDay) {
+        onSelectDay(this.state.selectedDates);
+      }
+    });
+  }
+
+  render() {
+    const { dataAttrs, isDaySelectableFn, locale, mods, selectionType } = this.props;
+    const { maxLimit, minLimit, renderDate, selectedDates } = this.state;
+
+    const restProps = getDataAttributes(dataAttrs);
+    const className = getClassNamesWithMods('ui-calendar', mods);
+
+    return (
+      <div className={className} {...restProps}>
+        <DaysPanel
+          isDaySelectableFn={isDaySelectableFn}
+          locale={locale}
+          maxDate={maxLimit}
+          minDate={minLimit}
+          onNavNextMonth={() => this.moveToMonth(CALENDAR_MOVE_TO_NEXT)}
+          onNavPreviousMonth={() => this.moveToMonth(CALENDAR_MOVE_TO_PREVIOUS)}
+          onSelectDay={dt => this.onSelectDay(dt)}
+          renderDate={renderDate}
+          selectedDates={selectedDates}
+          selectionType={selectionType}
+        />
+      </div>
+    );
+  }
+}
+
+Calendar.defaultProps = {
+  dataAttrs: {},
+  mods: [],
```

Comment:
unless you're not modifying it in the render function it's ok. but somebody else could not notice it and add some
`mods.push('some-mod')` and it will change mods across all instances of this component. It is not a bug in your code, but potential weak point and I consider as a "better practice" not to put objects\arrays into default props object